Session08/binarytree/solutions/BinaryTree.java

```diff
@@ -0,0 +1,79 @@
+abstract class BinaryTree {
+	protected TreeNode root;
+
+	/* While this constructor cannot be called directly,
+	   it is useful because the child classes can use it,
+	   by using super(), to instantiate the root node
+	 */
+	public BinaryTree(int rootValue) {
+		this.root = new TreeNode(rootValue);
+	}
+
+	public void inOrderTraversal() {
+		//TODO
+		inorderRec(root);
```

Comment:
I am assuming that `inOrderRec()` stands for `inOrderRecursive`. However, there is no way for me to know this, since it's not a convention. Please name it `inOrderRecursive`